airflow/utils/state.py

```diff
@@ -41,6 +41,21 @@ class State(object):
     UPSTREAM_FAILED = "upstream_failed"
     SKIPPED = "skipped"
 
+    task_states = (
+        SUCCESS,
+        RUNNING,
+        FAILED,
+        UPSTREAM_FAILED,
+        UP_FOR_RETRY,
+        QUEUED,
+    )
```

Comment:
Originally task_states was a tuple defined in the old /dag_stats (now
renamed to /task_stats) API endpoint. I made the decision to refactor it
into state.py.

On Wednesday, September 7, 2016, bolkedebruin <notifications@github.com>
wrote:

> In airflow/utils/state.py
> <https://github.com/apache/incubator-airflow/pull/1662#discussion_r77947094>
> :
>
> > @@ -41,6 +41,21 @@ class State(object):
> >      UPSTREAM_FAILED = "upstream_failed"
> >      SKIPPED = "skipped"
> >
> > +    task_states = (
> > +        SUCCESS,
> > +        RUNNING,
> > +        FAILED,
> > +        UPSTREAM_FAILED,
> > +        UP_FOR_RETRY,
> > +        QUEUED,
> > +    )
>
> I am missing scheduled here, and I dont understand the need for the
> separate set?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-airflow/pull/1662/files/f734d8a876b6a31abde57cedf5e44c69031f82b1#r77947094>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGYMpsN8UimiSujL_BsP5WeM8w0t7UOmks5qn5phgaJpZM4JLnBS>
> .
>